test/fast/Unit tests/nvm_install_binary_nosource

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+
+die () { echo "$@" ; cleanup ; exit 1; }
+
+
+\. ../../../nvm.sh
+
+nvm_install_binary() {
+    echo "NOSOURCE:${nosource-}"
+}
+
+OUTPUT="$(nvm install -b 8.0.0)"
+EXPECTED_OUTPUT='NOSOURCE:1'
+[ "${OUTPUT}" = "${EXPECTED_OUTPUT}" ] || die "No source binary flag is active and should have returned true. Instead it returned >${OUTPUT}<"
+
+OUTPUT="$(nvm install 8.0.0)"
+EXPECTED_OUTPUT='NOSOURCE:0'
+[ "${OUTPUT}" = "${EXPECTED_OUTPUT}" ] || die "No source binary flag is not active and should have returned false. Instead it returned >${OUTPUT}<"
+
+# Check if nosource stops downloading from source
+
+set +ex # needed for stder
+OUTPUT="$(nvm install -b 8.0.0 2>&1)"
```

Comment:
@ljharb Just not sure how would we make the binary download fail here. Maybe pass a version that did not have a binary version available

Reply:
we could override `nvm_binary_available` to return 1

Reply:
Yeah it looks cool now